mlrun/execution.py

```diff
@@ -248,6 +253,13 @@ def from_dict(
             self._log_level = spec.get("log_level", self._log_level)
             self._function = spec.get("function", self._function)
             self._parameters = spec.get("parameters", self._parameters)
+            if not self._iteration:
+                self._hyperparams = spec.get("hyperparams", self._hyperparams)
+                self._hyper_options = spec.get("hyper_options", self._hyper_options)
+                if isinstance(self._hyper_options, dict):
+                    self._hyper_options = HyperParamOptions.from_dict(
+                        self._hyper_options
+                    )
```

Comment:
Shouldn't you get from the `spec` ?

Reply:
the line above (`self._hyper_options = spec.get("hyper_options", self._hyper_options)`) read it from spec